src/NATS.Client.Core/INatsConnection.cs

```diff
@@ -113,10 +122,12 @@ public interface INatsConnection
     /// <remarks>
     /// if reply option's timeout is not defined then it will be set to NatsOpts.RequestTimeout.
     /// </remarks>
-    IAsyncEnumerable<NatsMsg<TReply?>> RequestManyAsync<TRequest, TReply>(
+    IAsyncEnumerable<NatsMsg<TReply>> RequestManyAsync<TRequest, TReply>(
         string subject,
         TRequest? data,
         NatsHeaders? headers = default,
+        INatsSerializer<TRequest>? requestSerializer = default,
+        INatsSerializer<TReply>? replySerializer = default,
```

Comment:
Which then also raises the question if it wouldn't be better (and feasible) to just use `Action<IBufferWriter<byte>, T>` and `Func<ReadOnlySequence<byte>, T>` so that clients don't have to implement the interface if they already have a suitable serializer that matches the delegates signature.

Reply:
Sounds ok but I'd like to see what @caleblloyd would say here.

Reply:
Splitting into `I(Nats)Serializer<T>` and `I(Nats)Deserializer<T>` makes sense to me

The `Action` / `Func` seems a little non-standard - it may map cleanly to some Serializer/Deseriaizers such as GRPC, but not others, such as JsonSerializer.  It should be trivial to use a helper function to create a static `I(Nats)Serializer<T>` and `I(Nats)Deserializer<T>` from an Action / Func though

Reply:
I agree we can split the interface but I'm leaning towards interfaces too rather than action/func approach. I believe interface would be more developer friendly in general.